s3/pkg/datastore/yig/storage/object.go

```diff
@@ -233,8 +232,58 @@ func (yig *YigStorage) Get(ctx context.Context, object *pb.Object, start int64,
 	return reader, nil
 }
 
+/*
+* @objectId: input object id which will be deleted.
+* Below is the process logic:
+* 1. check whether objectId is multipart uploaded, if so
+* retrieve all the object ids from multiparts and put them into gc.
+* or else, put it into gc.
+*
+ */
 func (yig *YigStorage) Delete(ctx context.Context, object *pb.DeleteObjectInput) error {
-	return errors.New("not implemented.")
+	// check multipart uploaded object.
+	if object.StorageMeta == "" {
```

Comment:
Could you please add comment to explain that's the scenario when object.StorageMeta equals ""?

Reply:
@sfzeng thanks for your suggestion, i have added comments for such check, pls review it again, thanks.